### DISCUSSION TITLE
Infinity Data Source - unable to fetch json data from url

### DISCUSSION THREAD

**Discussion by halomanoj** [grafana/grafana-infinity-datasource#909] [Q&A]
I have few url which give me data in json format. 
eg
1- https://wmidr_rest_service.apps.prod.na-5z.gap.jpmchase.net/monitoring/read_json/events/eventbridge_monitor_idr.json
2 - https://mis-reporting.test.aws.jpmchase.net/qrs-endpoint?api_name=dashboard_refresh_details

The 1st url works fine and I am able to fetch the data using infinity datasource
for the 2nd url i get errors - 
error getting data frame. error getting response from url https://mis-reporting.test.aws.jpmchase.net/qrs-endpoint?api_name=dashboard_refresh_details. no response received. Error: Get "https://mis-reporting.test.aws.jpmchase.net/qrs-endpoint?api_name=dashboard_refresh_details": dial tcp 100.78.189.187:443: connect: connection refused

The JSON data fetched from the 2nd url looks good and when i try that with inline option i am able to get it in tabular form.

JSON data from 2nd URL -
[
  {
    "eventGroupName": "NON_COLLATERAL_DASHBOARD_SF_NA",
    "eventGroupDescription": "Non-Collateral Dashboard SF",
    "eventSlaTime": "10:00:00",
    "appId": "8ddb6b82-8182-4df7-864b-9820e378b286",
    "taskName": "CreditMIS-Non-Collateral Dashboard SF-Ops",
    "cobDate": null,
    "refreshedDate": null,
    "status": null,
    "executionId": null,
    "createdDateTime": null,
    "updatedDateTime": null
  }
]

Is it due to the special characters in the 2nd url ( ? and = ). Please advise if i need to add some escape char before them.
Grafana Version - 9.4.7

**Comment by yesoreyeram** [comment]
> connect: connection refused

from the errors, it looks like some sort of connection issue from your grafana to the endpoint. If the APIs are behind proxy, make sure they are properly setup in the config. if not proxy, make sure the URL is accessible from the environment where your grafana is running via curl.
